resources/netregexes.ts

```diff
@@ -28,19 +28,40 @@ const keysThatRequireTranslation = [
 ];
 
 type ParseHelperField<
-  Type extends TriggerTypes,
-  Fields extends NetFieldsReverse[Type],
+  Type extends LogDefinitionTypes,
+  Fields extends LogDefinitionsReverse[Type],
   Field extends keyof Fields,
 > = {
   field: Fields[Field] extends string ? Fields[Field] : never;
   value?: string;
 };
 
-type ParseHelperFields<T extends TriggerTypes> = {
-  [field in keyof NetFieldsReverse[T]]: ParseHelperField<T, NetFieldsReverse[T], field>;
+type ParseHelperFields<T extends LogDefinitionTypes> = {
+  [field in keyof LogDefinitionsReverse[T]]: ParseHelperField<T, LogDefinitionsReverse[T], field>;
 };
 
-const parseHelper = <T extends TriggerTypes>(
+const defaultParams = <
+  T extends keyof typeof logDefinitions,
+>(type: T, include?: string[]): Partial<ParseHelperFields<T>> => {
+  include ??= Object.keys(logDefinitions[type].fields);
+  const params: { [index: number]: { field: string; value?: string } } = {};
+
+  for (const [prop, index] of Object.entries(logDefinitions[type].fields)) {
+    if (!include.includes(prop))
+      continue;
+    const param: { field: string; value?: string } = {
+      field: prop,
+    };
+    if (prop === 'type')
+      param.value = logDefinitions[type].type;
+
+    params[index] = param;
+  }
+
+  return params as unknown as Partial<ParseHelperFields<T>>;
```

Comment:
I don't see any way offhand without major surgery to this file to get this to work.

I think I'd like to land this, and then maybe take a second pass to see if I (or you or somebody else) can make it work? As far as I can tell, the issue is really about `index` as `prop` can be properly typed.  I wonder about passing in just an array of properties and not the indexes (with maybe prop: string overrides) and if that would be easier to type.